include/envoy/api/io_error.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/common/platform.h"
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+namespace Api {
+
+class IoError;
+
+// IoErrorCode::Again is used frequently. Define it to be a distinguishable address to avoid
+// frequent memory allocation of IoError instance.
+// If this is used, IoCallResult has to be instantiated with a deleter that does not
+// deallocate memory for this error.
+#define ENVOY_ERROR_AGAIN reinterpret_cast<Api::IoError*>(0x01)
```

Comment:
I think there was some consensus that this wasn't a good way of doing this. At minimum I'd like to continue this discussion, even though your PR is just moving this. Can you add a TODO comment here, and maybe create an issue for this so we don't lose track?

Reply:
Added a TODO in the code with a reference to #6037 since it looks like it is being handled there.